single_stage_detector/ssd/mlcube.py

```diff
@@ -0,0 +1,99 @@
+import os
+import yaml
+import typer
+import shutil
+import subprocess
+from pathlib import Path
+
+
+app = typer.Typer()
+
+
+class DownloadDataTask(object):
+
+    urls = {
+        "train2017": "http://images.cocodataset.org/zips/train2017.zip",
+        "val2017": "http://images.cocodataset.org/zips/val2017.zip",
+        "annotations": "http://images.cocodataset.org/annotations/annotations_trainval2017.zip"
+    }
+
+    @staticmethod
+    def download(cache_dir: Path, data_dir: Path, name: str) -> None:
+        os.makedirs(cache_dir, exist_ok=True)
+        os.makedirs(data_dir, exist_ok=True)
+
+        dest_dir = data_dir / name
+        if dest_dir.exists():
+            print(f"Data ({name}) has already been downloaded: {dest_dir}")
+            return
+
+        archive_name = Path(DownloadDataTask.urls[name]).name
+        cached_archive = cache_dir / archive_name
+        if not cached_archive.exists():
+            print(f"Data ({name}) is not in cache ({cached_archive}), downloading ...")
+            os.system(f"cd {cache_dir}; curl -O {DownloadDataTask.urls[name]};")
+
+        dest_archive = data_dir / archive_name
+        if cache_dir != data_dir:
+            print(f"Copying data from {cached_archive} to {dest_archive} ...")
+            shutil.copyfile(cached_archive, dest_archive)
+
+        print(f"Extracting archive ({archive_name}) ...")
+        os.system(f"cd {data_dir}; unzip {archive_name};")
```

Comment:
it would be helpful to have a consistency check here as well (presumably not an md5sum of every file, but some indication that the user has the right data?  For this case, perhaps the number of .jpgs?